data/settings.js

```diff
@@ -3014,186 +3110,305 @@ If set, allows message deliveries to exceed quota by this value.`
+allowing clients that move messages with IMAP COPY+EXPUNGE to Trash folder to
+temporarily exceed the quota.
 
-\`\`\`
-plugin {
-  quota = maildir:User quota
-  quota_rule = *:storage=1GB
-  # 10% of 1GB = 100MB
-  quota_rule2 = Trash:storage=+10%%
-  # 20% of 1GB = 200MB
-  quota_rule3 = Spam:storage=+20%%
+Example:
+\`\`\`[dovecot.conf]
+quota_storage_size = 1G
+namespace inbox {
+  mailbox Trash {
+    # 110% * 1G = 1100M
+    quota_storage_percentage = 110
+  }
 }
-\`\`\`
-
-Note that \`%\` is written twice to escape it, because [[variable]] are
-expanded in plugin section.
-
-[[link,userdb]] configuration may or may not require this escaping.
+\`\`\``
+	},
 
-Backend-specific configuration currently is used only with \`Maildir++\`
-quota backend. It means you can have the quota in Maildir++ format (e.g.
-\`10000000S\`).`
+	quota_storage_size: {
+		default: 'unlimited',
+		plugin: 'quota',
+		values: setting_types.UINT,
+		seealso: [
+			'[[setting,quota_storage_extra]]',
+			'[[setting,quota_storage_percentage]]',
+			'[[setting,quota_message_count]]',
+			'[[link,quota_root]]'
+		],
+		tags: [ 'storage_size_limits' ],
+		text: `
+Quota storage size limit for the [[link,quota_root]]. This value is still
+multiplied by [[setting,quota_storage_percentage]] and then increased by
+[[setting,quota_storage_extra]] to get the final value (in this mailbox or
+namespace). This is reported as the STORAGE limit in IMAP GETQUOTA commands.
+Using \`0\` as the value means the same as \`unlimited\`.`
 	},
 
 	quota_warning: {
 		plugin: 'quota',
-		values: setting_types.STRING,
+		values: setting_types.NAMED_LIST_FILTER,
 		seealso: [ '[[link,quota_warning_scripts]]' ],
+		tags: [ 'quota-warning' ],
 		text: `
-You can configure Dovecot to run an external command when user's quota
-exceeds a specified limit. Note that the warning is ONLY executed at the
-exact time when the limit is being crossed, so when you're testing you have
-to do it by crossing the limit by saving a new mail. If something else
-besides Dovecot updates quota so that the limit is crossed, the warning is
-never executed.
-
-Quota warning configuration has the following syntax:
+Create a new [[link,quota_warning_scripts,quota warning]]. The filter name
+refers to [[setting,quota_warning_name]] setting. The [[setting,execute]]
+setting is required to be specified inside the filter.
 
-\`\`\`
-quota_warning = <limit configuration> <quota-warning socket name> <parameters>
-\`\`\`
-
-\`limit_configuration\` is almost exactly same as for
-[[setting,quota]], with the exception of adding \`-\` before
-the value for "reverse" warnings where the script is called when quota
-drops below the value. Usually you want to use percents instead of absolute
-limits.
+The order of \`quota_warning\` filters in the configuration is important:
+Only the first warning that matches the rules is executed. This means
+configuring the highest limits first.`
```

Comment:
I would add more emphasis on this being a command: "This means you should/must/are highly recommended to configure the highest limits first." Or alternatively: "Configuring the highest limits first is paramount." Something like this.